### PR TITLE
Whitelist send_with_u8_array slice

### LINKED ISSUE #2014: `WebSocket.send_with_u8_array` should accept an immutable byte slice
Currently requires `&mut [u8]`

https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WebSocket.html#method.send_with_u8_array

### PR DESCRIPTION
**Pull request by chinedufn**
Fixes #2014

### CODE REVIEW

File: crates/webidl/src/lib.rs [hunk 1]
```diff
@@ -244,6 +244,8 @@ fn immutable_slice_whitelist() -> BTreeSet<&'static str> {
         "clearBufferfv",
         "clearBufferiv",
         "clearBufferuiv",
+        // WebSocket
+        "send_with_u8_array",
```
Comment (limira) [inline crates/webidl/src/lib.rs]: I am not sure but I *think* it requires *camelCase*?

Comment (alexcrichton) [reply crates/webidl/src/lib.rs]: Hm yeah it looks like the others are all camel-case here. @chinedufn mind testing this out locally to see if it had the desired effect on the method?

Comment (chinedufn) [reply crates/webidl/src/lib.rs]: Thanks! https://github.com/chinedufn/wasm-bindgen/commit/001f4b08844b12f8a110a1bcd248316f49b4c8f9#diff-266f7f5e4a4a2bbbf9377f3d19673606R95-R98

File: crates/web-sys/tests/wasm/whitelisted_immutable_slices.rs [hunk 2]
```diff
@@ -4,86 +4,89 @@
...
 #[wasm_bindgen(module = "/tests/wasm/element.js")]
 extern "C" {
     fn new_webgl_rendering_context() -> WebGlRenderingContext;
     fn new_webgl2_rendering_context() -> WebGl2RenderingContext;
+    fn new_websocket() -> WebSocket;
 // TODO: Add a function to create another type to test here.
 // These functions come from element.js
 }
 
-// TODO: Uncomment WebGlRenderingContext test. Every now and then we can check if this works
```
Comment (chinedufn) [inline crates/web-sys/tests/wasm/whitelisted_immutable_slices.rs]: Checking to see if these tests can pass in CI now sine it's been a year

Comment (chinedufn) [reply crates/web-sys/tests/wasm/whitelisted_immutable_slices.rs]: Nope

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**